File: Novels/packages/5-mobile-ios/StoryReader/Model/Tag.swift
import Foundation

// Tag model - Supports hierarchical tags
struct Tag: Codable, Identifiable {
    let id: String
    let userId: String
    let name: String
    let color: String?
    let icon: String?
    let parentTagId: String?
    let parentTag: Tag?
    let childTags: [Tag]?
    let createdAt: Date
    let updatedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name
        case color
        case icon
        case parentTagId = "parent_tag_id"
        case parentTag = "parent_tag"
        case childTags = "child_tags"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Tag.kt
package com.storysphere.storyreader.model

data class Tag(
    val id: String,
    val userId: String,
    val name: String,
    val color: String? = null,
    val icon: String? = null,
    val parentId: String? = null, // For hierarchical tags
    val order: Int = 0,
    val createdAt: Long = System.currentTimeMillis(),
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
    val lastSyncedAt: Long? = null
)

